types/ndarray/index.d.ts

```diff
@@ -33,8 +34,7 @@ declare namespace ndarray {
         pick(...args: Array<number | null>): NdArray<D>;
         T: NdArray<D>;
     }
-
-    type Data<T = any> = T[] | TypedArray;
+    type Data<T = any> = T extends bigint ? T[] | TypedArray | BigInt64Array | BigUint64Array : T[] | TypedArray;
```

Comment:
Thanks for the review & feedback!

I avoided this because `Value<Data<number>>` is `bigint | number`, and this would break type signatures that don't provide a generic parameter for `ndarray.NdArray`:

```typescript
function getFirstValue(arr: ndarray.NdArray): number {
    return arr.get(0); // Type 'number | bigint' is not assignable to type 'number'
}
```

The conditional type in `Data` makes it so that `Value<Data<bigint>>` == `bigint | number` whereas `Value<Data<number>>` remains `number`; the goal was to widen the default type based on whether `T` includes `bigint`.

I think avoiding the conditional type is more correct, but this was a decision to avoid breaking changes, but I am in support of making the changes ultimately. Also, I don't think there is a need for conditional types in `Value`. All types captured by `Data<any>` are indexible, so value can simply be,

```typescript
type Value<D extends Data> = D[number];

Value<BigUint64Array | ({ a: 'b' } | boolean)[] > // bigint | boolean | { a: 'b' }
```